src/gfx/gizmos/rotation_gizmo.hpp

```diff
@@ -111,10 +115,17 @@ struct RotationGizmo : public IGizmo, public IGizmoCallbacks {
       float3 radiusVec = hitPoint - selectionDisc.center;
 
       // solnA using the tangent to the disc and the normal of the disc to form the plane of rotation
+      // the problem with this solution is that in the case that the ring is along a plane that is near perpendicular,
+      // the distance along the tangent moved can easily grow exponentially, causing the object to "spin-out" or rotate too much
       dragStartPoint = hitPoint;
       dragTangentDir = linalg::normalize(linalg::cross(selectionDisc.normal, radiusVec));
 
-      // solnB using the plane normal to the eye location as the plane of rotation, where tangent is projected onto this plane
+      // solnB using the plane that is normal to the eye location as the plane of rotation (from the hitPoint), 
+      // where tangent is projected onto this plane. this solution is not using true screen space, but the plane should be
+      // almost parallel to the viewport, such that we do not get the same spin-out issue
+      // not sure if possible to get true distane in screen space because information about the true normal vector to the
+      // viewport is not visible to the gizmo (need to be calculated through e.g. InputContext::updateView?)
+      // builds upon the results of solnA rather than using the 2 lines above as-is
       dragNormalDir = linalg::normalize(context.eyeLocation - hitPoint);
       dragTangentDir = linalg::normalize(dragTangentDir - linalg::dot(dragTangentDir, dragNormalDir) * dragNormalDir);
```

Comment:
solnA is basically raycasting onto a plane that the disc lies on, which means that if the disc is at an angle almost perpendicular to the viewport, the distance along the tangent to the hitPoint will grow exponentially and cause the object to spin-out

solnB is finding the tangent to the hitPoint along the disc, then projecting that tangent onto a plane whose normal vector goes from the hitPoint to the eyeLocation, so that the plane will never be too perpendicular. 

The tangent is to adjust the direction that the user should drag the handle along, to the point of the disc clicked on.

Reply:
@guusw Currently theres some limitations of the current solution (solnB) having not true screen space becomes more obvious when the gizmo has been translated to the corner of the screen/viewport, where the amount of rotation per unit mouse distance moved in either direction becomes skewed.

In the future, if possible, I think this would likely be better if it can be linked to the movement of cursorPosition to overcome this issue, and possibly have the same feature as Unity where the user can wrap the cursor around the screen when it goes out of the screen while holding the RotationGizmo handle, so that the user may keep dragging and rotating without running out of screen space to drag the cursor.

Reply:
Now linked to a plane parallel to the screen space, but not linked to movement of cursorPosition, which may be a bit more efficient